pkg/proxy/apis/config/validation/validation.go

```diff
@@ -69,14 +71,20 @@ func Validate(config *kubeproxyconfig.KubeProxyConfiguration) field.ErrorList {
 	allErrs = append(allErrs, validateHostPort(config.MetricsBindAddress, newPath.Child("MetricsBindAddress"))...)
 
 	if config.ClusterCIDR != "" {
-		if config.FeatureGates[string(kubefeatures.IPv6DualStack)] {
-			cidrs := strings.Split(config.ClusterCIDR, ",")
-			for _, cidr := range cidrs {
-				if _, _, err := net.ParseCIDR(cidr); err != nil {
-					allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), cidr, "must be a valid CIDR block (e.g. 10.100.0.0/16 or FD02::0:0:0/96)"))
-				}
+		cidrs := strings.Split(config.ClusterCIDR, ",")
+
+		switch {
+		// if DualStack and more than one cidr validate if there is at least one of each IP family
+		case utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) && len(cidrs) > 1:
```

Comment:
better to move the output of utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) to a local variable above called - e.g. dualStackEnabled.